include/ruby/assert.h

```diff
@@ -31,6 +31,10 @@
 # define RUBY_DEBUG 0
 #endif
 
+#if RUBY_DEBUG > 0 && defined(NDEBUG)
+#warning NDEBUG is ignored because RUBY_DEBUG>0.
```

Comment:
I don't think it works for MSVC.

Reply:
```suggestion
#if defined(_MSC_VER)
# pragma message("NDEBUG is ignored because RUBY_DEBUG>0.")
#elif defined(__GNUC__)
# pragma GCC warning "NDEBUG is ignored because RUBY_DEBUG>0."
#else 
# error NDEBUG is ignored because RUBY_DEBUG>0.
#endif
```